### ISSUE TITLE
[BUG] Textures not loading correctly

### ISSUE THREAD

**Issue by trashuj** [amethyst/amethyst#2280]
## Description
Textures are not appearing or get mixed up.

## Reproduction Steps
https://github.com/trashuj/crash_amethyst_investigation
Click on exit/join to go back and forth.

## What You Expected to Happen
If you run the code above, and you click on exit/join to go back and forth, you should see the equivalent of the screenshot.png (in the repo) every time. 

## What Actually Happened
Sometimes the textures get mixed up, or they dont appear at all :s

## Screenshots and Logs
What we expect
![what_we_want](https://user-images.githubusercontent.com/9165398/81404408-affb5600-9135-11ea-9134-8fbdb3c1c46c.png)

What we get
![what_we_get](https://user-images.githubusercontent.com/9165398/81404432-bbe71800-9135-11ea-92c6-1138db10d213.png)

## Your Environment
1. Operating System: Windows 10
2. Operating System Version: pro
3. Version of Amethyst: 0.15.0
 I tested and got the same issue on linux ubuntu

## Additional Context
if someone could please at least just run the program to confirm it's not happening only to me :s, I would appreciate!

investigation related by Intel 
https://github.com/amethyst/amethyst/pull/2258

**Comment by trashuj** [comment]
Btw intel is right, if i keep and reuse the handles it works fine, but
1) i have to move everything from ui ron file in code
2) eventually the gc will run out of memory

**Comment by ab0v3g4me** [comment]
This was merged so I'm closing this one.

**Comment by azriel91** [comment]
This issue still exists, and I think #2258 actually caused it (this issue was opened after #2258 got merged).
I'll look into what's causing it.

**Comment by azriel91** [comment]
Okay, so before #2258, materials and textures weren't being freed when they were no longer in use, so #2258 fixes that. That exposed another issue where, when asset handles that are reused, there seems to be a discrepancy between the `Handle` id, and the `TextureSub`'s internal slotmap id.

Still figuring that part out.

**Comment by azriel91** [comment]
https://github.com/amethyst/amethyst/pull/2339 should fix it -- at least in my game, the textures that I load / unload on the fly work correctly now.
